packages/react-aria-components/src/FileTrigger.tsx

```diff
@@ -62,7 +67,9 @@ function FileTrigger(props: FileTriggerProps, ref: ForwardedRef<HTMLInputElement
         accept={acceptedFileTypes?.toString()}
         onChange={(e) => onSelect?.(e.target.files)}
         capture={defaultCamera}
-        multiple={allowsMultiple} />
+        multiple={allowsMultiple} 
+        // @ts-expect-error
+        webkitdirectory={UNSAFE_directory ? '' : undefined} />
```

Comment:
this implementation uses an empty string for the webkitdirectory attribute instead of a boolean value because it does not function correctly with boolean. 

ref: https://github.com/facebook/react/issues/3468#issuecomment-448336672